### PR TITLE
Fix Windows builds

### PR DESCRIPTION
**Pull request by jblazquez**
The library doesn't build on Windows due to a small number of errors, mostly missing definitions for a few functions.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 cmake_minimum_required(VERSION 3.1)
 # XXX using 0.1.90 instead of 0.1.0-DEV
-project(nghttp2 VERSION 0.1.90)
+project(ngtcp2 VERSION 0.1.90)
```
Comment (jblazquez) [inline CMakeLists.txt:27]: This fixes the name of the Visual Studio solution when using `cmake -G "Visual Studio ..."`.

File: lib/ngtcp2_net.h [hunk 2]
```diff
@@ -0,0 +1,91 @@
+/*
```
Comment (jblazquez) [inline lib/ngtcp2_net.h:1]: This file is copied from the nghttp2 project.

**Comment by alagoutte** [comment]
What step/command for build with Windows ?

Can you look to add build system for appveyor (or travis) ?

**Comment by jblazquez** [comment]
Hi @alagoutte,

Using CMake the commands to build on Windows are pretty simple. With this change applied, you can build it in four different ways. First, open a **x64 Native Tools Command Prompt for VS 2017** that you can get by installing the [Visual Studio 2017 Command Line Tools](https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=vs-2017#developer-command-prompt-shortcuts), then:

```
mkdir build
cd build
```

Now you have four options:

1. Use MSBuild:

```
cmake -G "Visual Studio 15 2017" ..
cmake --build .
```

2. Use the Visual Studio build system:

```
cmake -G "Visual Studio 15 2017" ...
devenv ngtcp2.sln /build
```

3. Use NMake:

```
cmake -G "NMake Makefiles" ..
nmake
```

4. Use [Ninja](https://ninja-build.org/):

```
cmake -G "Ninja" ...
ninja
```

I also added AppVeyor integration based on your configuration from the libnghttp2 project: https://ci.appveyor.com/project/jblazquez/ngtcp2/builds/22492075

**Comment by tatsuhiro-t** [comment]
Thank you for PR.  Merged now.

**Pull request merged**